--- /sites/all/modules/taxonomy_menu/taxonomy_menu.module	Thu Dec 23 16:40:54 2010
+++ /sites/all/modules/taxonomy_menu/taxonomy_menu.module	Fri Jan 14 01:10:06 2011
@@ -540,9 +540,7 @@
 function _taxonomy_menu_save($item) {
   $insert = TRUE;
   //create the path.
-  //use url to create he inital path
-  //we need to remove the first '/' so menu_link_save will work correctly
-  $path = ltrim(taxonomy_menu_create_path($item['vid'], $item['tid']), '/');
+  $path = taxonomy_menu_create_path($item['vid'], $item['tid']);
   // get the parent mlid: this is either:
   // - the parent tid's mlid
   // - the vocab menu item's mlid
@@ -679,9 +677,7 @@
     $path = "taxonomy/term/$end";
   }
   else {
-    $path = url('taxonomy/term/' . $tid);
-    // Url adds a '/' to the beginning so remove it now.
-    $path = substr($path, 1);
+    $path = 'taxonomy/term/' . $tid;
     if (variable_get('taxonomy_menu_display_descendants_' . $vid, FALSE)) {
       // Use 'all' at the end of the path
       if (variable_get('taxonomy_menu_end_all_' . $vid, FALSE)) {

